packages/aleph-holders/src/domain/worker.ts

```diff
@@ -11,62 +11,64 @@ import {
   EthereumLogIndexerWorkerDomainI,
   EthereumParsedLog,
 } from '@aleph-indexer/ethereum'
-import { BscLogIndexerWorkerDomainI, BscParsedLog } from '@aleph-indexer/bsc'
+import { 
+  BscLogIndexerWorkerDomainI, 
+  BscParsedLog 
+} from '@aleph-indexer/bsc'
 import {
   EventType,
-  ERC20TransferEventQueryArgs,
-  ERC20TransferEvent,
+  TransferEventQueryArgs,
+  TransferEvent,
   BalanceQueryArgs,
   Balance,
+  SolanaBalance,
 } from '../types.js'
 import {
-  ERC20TransferEventDALIndex,
-  createERC20TransferEventDAL,
-} from '../dal/erc20TransferEvent.js'
-import { EventParser } from './parser.js'
+  TransferEventDALIndex,
+  createTransferEventDAL,
+} from '../dal/transfer.js'
+import { EthereumEventParser } from './ethereumParser.js'
 import { BalanceDALIndex, createBalanceDAL } from '../dal/balance.js'
 import {
   blockchainDeployerAccount,
   blockchainTotalSupply,
 } from '../utils/index.js'
+import { SolanaIndexerWorkerDomainI, SolanaParsedInstructionContext } from '@aleph-indexer/solana'
 
 export default class WorkerDomain
   extends IndexerWorkerDomain
-  implements EthereumLogIndexerWorkerDomainI, BscLogIndexerWorkerDomainI
-{
+  implements
+    EthereumLogIndexerWorkerDomainI,
+    BscLogIndexerWorkerDomainI,
+    SolanaIndexerWorkerDomainI {
+
   constructor(
     protected context: IndexerDomainContext,
-    protected erc20TransferEventDAL = createERC20TransferEventDAL(
-      context.dataPath,
-    ),
+    protected ethParser = new EthereumEventParser(),
+    protected transferEventDAL = createTransferEventDAL(context.dataPath),
     protected balanceDAL = createBalanceDAL(context.dataPath),
-    protected parser = new EventParser(),
   ) {
     super(context)
   }
 
-  async onNewAccount(config: AccountIndexerRequestArgs): Promise<void> {
-    const { blockchainId: blockchain, account: contract } = config
+  async onNewAccount(
+    config: AccountIndexerConfigWithMeta<SolanaBalance>,
+  ): Promise<void> {
+    const { blockchainId: blockchain, account, meta } = config
     const { instanceName } = this.context
 
-    const deployer = blockchainDeployerAccount[blockchain]
-    const accountBalance = await this.balanceDAL.get([blockchain, deployer])
-
-    console.log(
-      'Account indexing',
-      instanceName,
-      blockchain,
-      contract,
-      deployer,
-      accountBalance,
-    )
+    console.log('Account indexing', instanceName, blockchain, account)
 
-    // @note: Init the initial supply if it is the first it run
-    if (!accountBalance) {
-      const balance = blockchainTotalSupply[blockchain].toString('hex')
-      await this.balanceDAL.save({ blockchain, account: deployer, balance })
+    if (!meta) {
```

Comment:
Why not on `meta`?

Reply:
Token contracts are indexed without any meta, inside the condition is executed code related to token contracts

Reply:
Would be good to comment on that. Had no idea.